include/class.ticket.php

```diff
@@ -2903,6 +2914,17 @@ function postCannedReply($canned, $message, $alert=true) {
     function postReply($vars, &$errors, $alert=true, $claim=true) {
         global $thisstaff, $cfg;
 
+        // Validate Form Token to prevent form resubmissions
+        if (!$vars['form_token'] || !$vars['lockCode'])
+            $errors['err'] = __('Invalid Form Token');
+
+        if (!$vars['token_validated'] && !Lock::validateToken($vars['form_token'], $vars['lockCode'])) {
```

Comment:
As mentioned above - resubmission prevention should be at form level. In this case it should be part of `Response Form`.